system/jlib/jstats.cpp

```diff
@@ -1217,7 +1218,13 @@ unsigned __int64 mergeStatisticValue(unsigned __int64 prevValue, unsigned __int6
             return newValue;
         else
             return prevValue;
+    case StatsMergeFirst:
+        if (newValue && prevValue > newValue)
```

Comment:
needs to be 
if (newValue && ((prevValue > newValue) || !prevValue))
(as you have in store_first)